src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/Dashboard.xaml.cs

```diff
@@ -340,6 +342,22 @@ private void Commit()
                 var notificationService = _model.Session.Workspace.Services.GetService<INotificationService>();
                 notificationService.SendNotification(ex.Message, title: EditorFeaturesResources.Rename, severity: NotificationSeverity.Error);
             }
+            catch (Exception ex) when (FatalError.ReportAndCatch(ex))
+            {
+                // Show a nice error to the user via an info bar
+                var errorReportingService = _model.Session.Workspace.Services.GetService<IErrorReportingService>();
```

Comment:
Is this allowed? Nothing else in this assembly uses the info bar (see above - it just shows a message box) and I'm always nervous being first. Is this some violation of layering?

Reply:
No layering violation here: I'm surprised this would be the first. I can imagine that the editor features layer has generally been a place where we _would_ show a dialog on the error, and we use the IErrorReportingService for reporting of background errors.

That said now -- should this be moved into the session.Commit() method? Do we have proper alerting for errors if I commit a session by pressing enter or by starting another session?

Reply:
There is certainly no error handling in the Return command handler, but maybe the platform handles errors from command handlers differently?

@sharwell in the issue you mentioned the fix should go in Apply_Click, was there extra context on that?

Reply:
Ping @sharwell

Reply:
I'm not Sam, but I think this location is fine. We're assert to that Commit should not throw an exception at all, and this is how we're handling it. If other logic is added to the click handler besides commit, it may have different error handling logic requirements.